### PR TITLE
Add Compilation Changes

### PR DESCRIPTION
**Pull request by Gonmeso**
This PR aims to make the changes regarding to the compilation included in new versions of the API.

## Current Behaviour
All models were transpiled, but not compilation check at transpilation

## New behaviour
* After transpilation the model is checked for incompatible operators not yet supported in the Transpiler/Orion
* If all the operators are supported the  model is compiled
* If compilation is successful the sierra file is saved at Giza Platform
* Deployments can be deployed without the need of sierra files for fully supported models
* A version which is not fully supported now will be in `PARTIALLY_SUPPORTED` state

### CODE REVIEW

File: giza/frameworks/cairo.py [hunk 1]
```diff
@@ -296,13 +310,18 @@ def transpile(
                     time.sleep(10)
                 elif version.status == VersionStatus.COMPLETED:
                     echo.debug("Transpilation is ready, downloading! ✅")
-                    cairo_model = client.download(model.id, version.version)
+                    echo(
+                        "Transpilation is fully compatible. Version compiled and Sierra is saved at Giza ✅"
+                    )
+                    break
+                elif version.status == VersionStatus.PARTIALLY_SUPPORTED:
+                    echo.warning("🔎 Transpilation is partially supported!")
```
Comment (franalgaba) [inline giza/frameworks/cairo.py:318]: I would add a more verbose message saying. Its partially support which means some operators of your model are not supported. Go to Orion compatibility to check all the compatible ones and contribute to orion to have its support

File: giza/frameworks/cairo.py [hunk 2]
```diff
@@ -120,9 +120,9 @@ def prove(
 
 
 def deploy(
-    data: str,
     model_id: int,
     version_id: int,
+    data: Optional[str] = None,
```
Comment (franalgaba) [inline giza/frameworks/cairo.py:125]: Whats this? Docstring needed?

**Review by franalgaba — CHANGES_REQUESTED** [review]

File: giza/frameworks/cairo.py (hunk 2 above)
Comment (Gonmeso) [reply giza/frameworks/cairo.py:125]: Data is the sierra file, but just left it as `data`will add docstring

Comment (Gonmeso) [reply giza/frameworks/cairo.py:125]: It is already int he docstring:

`data: main SIERRA file`

**Review by franalgaba — APPROVED** [review]

**Pull request merged**